classes/announcement/PKPAnnouncementDAO.inc.php

```diff
@@ -370,16 +372,18 @@ function &getNumAnnouncementsNotExpiredByAssocId($assocType, $assocId, $numAnnou
 	}
 
 	/**
-	 * Retrieve most recent announcement by Assoc ID.
+	 * Retrieve most recent published announcement by Assoc ID.
 	 * @param $assocType int
 	 * @return Announcement
 	 */
-	function &getMostRecentAnnouncementByAssocId($assocType, $assocId) {
+	function &getMostRecentPublishedAnnouncementByAssocId($assocType, $assocId) {
```

Comment:
This is for OJS 2.4.x, and I'd rather not change reference semantics until 3.0. While technically I suspect we no longer have good PHP4 compatibility, there's no sense in disturbing stable code.